core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java

```diff
@@ -308,6 +334,41 @@ private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec, int right
     }
   }
 
+  private boolean checkWindowFunctionContainOrderBy(SqlCall call) {
+    return !((SqlWindow) call.operand(1)).getOrderList().getList().isEmpty();
+  }
+
+  private void unparseSqlWindow(SqlWindow sqlWindow, SqlWriter writer, SqlCall call) {
+    final SqlWindow window = sqlWindow;
+    writer.print("OVER ");
+    SqlCall operand1 = call.operand(0);
+    final SqlWriter.Frame frame =
+            writer.startList(SqlWriter.FrameTypeEnum.WINDOW, "(", ")");
+    if (window.getRefName() != null) {
+      window.getRefName().unparse(writer, 0, 0);
+    }
+    if (((SqlBasicCall) call.operand(0)).getOperandList().isEmpty()
+            && ((SqlWindow) call.operand(1)).getPartitionList().size() == 0
+                && ((SqlWindow) call.operand(1)).getOrderList().size() == 0) {
+      writer.endList(frame);
+    } else if (window.getOrderList().size() == 0) {
+      if (window.getPartitionList().size() > 0) {
+        writer.sep("PARTITION BY");
+        final SqlWriter.Frame partitionFrame = writer.startList("", "");
+        window.getPartitionList().unparse(writer, 0, 0);
+        writer.endList(partitionFrame);
+      }
+      writer.print("ORDER BY ");
+      SqlNode operand2 = operand1.operand(0);
+      operand2.unparse(writer, 0, 0);
+
+      writer.print("ROWS BETWEEN ");
```

Comment:
Please reduce function calls. Please use sep() for separators and print() to write strings